packages/lesswrong/server/callbacks/commentCallbacks.ts

```diff
@@ -1,667 +1,141 @@
- */
-async function forumEventSideEffects({ comment, forumEventMetadata }: { comment: DbComment; forumEventMetadata: ForumEventCommentMetadata; }) {
-  if (forumEventMetadata.eventFormat === "STICKERS") {
-    const sticker = forumEventMetadata.sticker
-
-    if (!comment.forumEventId) {
-      throw new Error("Comment must have forumEventId")
-    }
-
-    const {_id, x, y, theta, emoji} = sticker ?? {};
-
-    if (!sticker || !_id || !x || !y || !theta) {
-      throw new Error("Must include sticker")
-    }
-
-    if (!emoji) {
-      throw new Error("No emoji selected")
-    }
-
-    const forumEventId = comment.forumEventId;
-    const stickerData = {_id, x, y, theta, emoji, commentId: comment._id, userId: comment.userId};
-
-    await new ForumEventsRepo().addSticker({ forumEventId, stickerData });
-    captureEvent("addForumEventSticker", {
-      forumEventId,
-      stickerData,
-    });
-  }
 }
 
-getCollectionHooks("Comments").newSync.add(async (comment: DbComment, _, context: ResolverContext) => {
-  if (comment.forumEventMetadata) {
-    // Side effects may need to reference the comment, so set the _id now
-    comment._id = comment._id || randomId();
-    await forumEventSideEffects({ comment, forumEventMetadata: comment.forumEventMetadata });
-  }
+async function commentEditSync(modifier: MongoModifier<DbComment>, comment: DbComment, _0: DbUser | null, _1: DbComment, props: UpdateCallbackProperties<'Comments'>): Promise<MongoModifier<DbComment>> {
+  modifier = await moveToAnswers(modifier, comment, props.context);
+  modifier = await handleForumEventMetadataEdit(modifier, comment, props.context);
+
+  return modifier;
+}
+
+async function commentUpdateAfter(comment: DbComment, props: UpdateCallbackProperties<'Comments'>): Promise<DbComment> {
+  invalidatePostOnCommentUpdate(comment);
+  comment = await updateDescendentCommentCountsOnEdit(comment, props);
+
+  // notifyUsersAboutMentions
+
+  // 12 countOfReferenceCallbacks
+
   return comment;
-});
-
-getCollectionHooks("Comments").editSync.add(async (modifier, comment: DbComment) => {
-  const newMetadata = modifier.$set?.forumEventMetadata;
-  if (newMetadata && !isEqual(comment.forumEventMetadata, newMetadata)) {
-    await forumEventSideEffects({ comment, forumEventMetadata: newMetadata });
-  }
-  return modifier
-});
+}
+
+async function commentUpdateAsync(props: UpdateCallbackProperties<'Comments'>) {
+  await updatedCommentMaybeTriggerReview(props);
+  await updateUserNotesOnCommentRejection(props);
+  await checkModGPTOnCommentUpdate(props);
+}
+
+async function commentEditAsync(comment: DbComment, oldComment: DbComment, currentUser: DbUser | null, collection: CollectionBase<'Comments'>, props: UpdateCallbackProperties<'Comments'>) {
+  await commentsAlignmentEdit(comment, oldComment, props.context);
+  // There really has to be a currentUser here.
+  await commentsEditSoftDeleteCallback(comment, oldComment, currentUser!, props.context);
```

Comment:
I think this is an occasion where there was another type error due to Typescript's loose function parameter variance - `CommentsEditSoftDeleteCallback` used to accept a `DbUser`, even though `edit.async` callbacks only get a `DbUser | null`, according to [mutationCallbacks](https://github.com/ForumMagnum/ForumMagnum/blob/master/packages/lesswrong/server/mutationCallbacks.ts#L64)

Reply:
(However, in this case, it was probably fine; in practice we can't have a null `currentUser` when editing comments.)